### PR TITLE
Reveal in files instead of Finder

### LINKED ISSUE #12776: Reveal in finder refer to mac os and on linux or other
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In Linux we don't have finder we have a file explorer and in Linux when i see reveal in finder i cannot understand what it mean because it look like i must do some search and i just want to open the folder where the file is

### Environment

Linux

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Maybe Open in file explorer and stay with Finder for the mac user if its easy for mac users

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by francesco-gaglione**
fixes: #12776 

Release Notes:

- Renamed `editor::RevealInFinder` to `editor::RevealInFileManager`

### CODE REVIEW

**Comment by francesco-gaglione** [comment]
@cla-bot check

**Comment by mrnugget** [comment]
If we're doing a rename, I think we should use the generic term [file manager](https://en.wikipedia.org/wiki/File_manager). Finder/Files are specific application names.

**Comment by francesco-gaglione** [comment]
@mrnugget working on it

**Comment by francesco-gaglione** [comment]
#13322 
Reference

**Comment by mikayla-maki** [comment]
I don't think we should do a rename, I'd prefer if we added actions for each operating system, and then used `cfg!()`to switch between which actions we register and use in the UI.

**Review by mikayla-maki — CHANGES_REQUESTED** [review]
I don't think we should do a rename, I'd prefer if we added actions for each operating system, and then used `cfg!()`to switch between which actions we register and use in the UI.

**Comment by francesco-gaglione** [comment]
@mikayla-maki wrapped actions with your suggestion. Check it if it is enough. Thank you.

File: crates/editor/src/mouse_context_menu.rs [hunk 1]
```diff
@@ -90,7 +91,12 @@ pub fn deploy_context_menu(
                 .action("Copy", Box::new(Copy))
                 .action("Paste", Box::new(Paste))
                 .separator()
-                .action("Reveal in Finder", Box::new(RevealInFinder))
+                .when(cfg!(target_os = "macos"), |builder| {
+                    builder.action("Reveal in Finder", Box::new(RevealInFileManager))
+                })
+                .when(cfg!(not(target_os = "macos")), |builder| {
+                    builder.action("Reveal in File Manager", Box::new(RevealInFileManager))
+                })
```
Comment (JunkuiZhang) [inline crates/editor/src/mouse_context_menu.rs:99]: How about compilation check?
```suggestion
                action(ACTION_NAME, Box::new(RevealInFileManager))
```

And somewhere:
```rust
#[cfg(target_os = "macos")]
const ACTION_NAME: &str = "Reveal in Finder";
```

Comment (JunkuiZhang) [reply crates/editor/src/mouse_context_menu.rs:99]: This is just my personal thought :) . Personally, I prefer compiling checks over runtime checks. The final decision is yours.

Comment (francesco-gaglione) [reply crates/editor/src/mouse_context_menu.rs:99]: @JunkuiZhang I am implementing your suggestion. I just have one question. I don't know all the source code well yet, but I guess the action name can be inserted in some crate or some specific place to be then used throughout the source code. Any suggestions?

For what I've seen now in the source code I could put it in `editor/src/actions.rs`, does that work for you? Or would you prefer that I put it in the same file where it is used for the moment?

Comment (JunkuiZhang) [reply crates/editor/src/mouse_context_menu.rs:99]: Looks good to me :)

Comment (mikayla-maki) [reply crates/editor/src/mouse_context_menu.rs:99]: I think I disagree with this personally, the original version says what's going on very explicitly. That said, neither implementation would be a blocker :)

**Review by mikayla-maki — APPROVED** [review]

File: crates/editor/src/mouse_context_menu.rs (hunk 1 above)
Comment (francesco-gaglione) [reply crates/editor/src/mouse_context_menu.rs:99]: So in this case we can consider this issue/PL closed? @mikayla-maki

**Comment by francesco-gaglione** [comment]
This pr can be merged or I have to solve conflicts first?

**Comment by mrnugget** [comment]
> This pr can be merged or I have to solve conflicts first?

I think rebasing on main (or merging it in) and fixing the conflicts would be great.

**Comment by francesco-gaglione** [comment]
@mrnugget done it. so we can merge?

File: assets/keymaps/default-macos.json [hunk 2]
```diff
@@ -402,7 +402,7 @@
       "alt-cmd-]": "editor::UnfoldLines",
       "ctrl-space": "editor::ShowCompletions",
       "cmd-.": "editor::ToggleCodeActions",
-      "alt-cmd-r": "editor::RevealInFinder",
+      "alt-cmd-r": "editor::RevealInFileManager",
```
Comment (mrnugget) [inline assets/keymaps/default-macos.json:405]: Do we have to do this still? I think one point behind @mikayla-maki saying that we don't rename is that we also don't break the keybindings.

File: docs/src/key-bindings.md [hunk 3]
```diff
@@ -174,7 +174,7 @@ See the [tasks documentation](/docs/tasks#custom-keybindings-for-tasks) for more
 #### Editor
 
 | **Command**                      | **Target** | **Default Shortcut**            |
-| -------------------------------- | ---------- | ------------------------------- |
```
Comment (pzurek) [inline docs/src/key-bindings.md:177]: This change makes columns inconsistent. 2nd and 3rd still have spaces. Generally it seems like a convention to have the spaces there so how would you feel about undoing this change?

File: docs/src/key-bindings.md [hunk 4]
```diff
@@ -465,7 +465,7 @@ See the [tasks documentation](/docs/tasks#custom-keybindings-for-tasks) for more
 #### Project Panel
 
 | **Command**             | **Target**    | **Default Shortcut**  |
-| ----------------------- | ------------- | --------------------- |
```
Comment (pzurek) [inline docs/src/key-bindings.md:468]: Ditto.

File: assets/keymaps/default-macos.json (hunk 2 above)
Comment (francesco-gaglione) [reply assets/keymaps/default-macos.json:405]: Fixing

File: docs/src/key-bindings.md (hunk 3 above)
Comment (francesco-gaglione) [reply docs/src/key-bindings.md:177]: Sorry my editor autocompleted it fixing

**Comment by francesco-gaglione** [comment]
I don't have a mac can anyone test before merging?

File: assets/keymaps/default-macos.json [hunk 5]
```diff
@@ -402,7 +402,7 @@
       "alt-cmd-]": "editor::UnfoldLines",
       "ctrl-space": "editor::ShowCompletions",
       "cmd-.": "editor::ToggleCodeActions",
-      "alt-cmd-r": "editor::RevealInFileManager",
+      "alt-cmd-r": "editor::RevealInFinder",
```
Comment (pzurek) [inline assets/keymaps/default-macos.json:405]: This causes the following error on macOS:

```
cargo run
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.66s
     Running `target/debug/zed`
[2024-07-06T08:14:54+12:00 ERROR settings] invalid binding value for keystroke alt-cmd-r, context Some("Editor")

Caused by:
    no action type registered for editor::RevealInFinder
[2024-07-06T08:14:54+12:00 ERROR settings] invalid binding value for keystroke alt-cmd-r, context Some("Editor")

Caused by:
    no action type registered for editor::RevealInFinder
```

**Comment by mrnugget** [comment]
I tested on Linux and on macOS. On Linux everything works as expected. On macOS the action is now called "Reveal in File Manager".

I think @mikayla-maki's idea was that we define two actions: `RevealInFinder` and `RevealInFileManager` and only use one of two, depending on the target dir. (Also: `reveal_in_finder` is still the method name.)

The problem is that for the `vim.json` keybindings we have to pick one over the other and we can't do a runtime check of `vim.enabled` to define one or the other action.

Sooooo, should we go with the rename then, @mikayla-maki? :D

**Review by mikayla-maki — APPROVED** [review]
@mrnugget Agreed! Thanks for bringing that issue to my attention :)

**Pull request merged**